files/zh-cn/web/api/service_worker_api/using_service_workers/index.md

```diff
@@ -5,436 +5,446 @@ slug: Web/API/Service_Worker_API/Using_Service_Workers
-3. `scope` 参数是选填的，可以被用来指定你想让 service worker 控制的内容的子目录。在这个例子里，我们指定了 `'/sw-test/'`，表示 app 的 origin 下的所有内容。如果你留空的话，默认值也是这个值，我们在指定只是作为例子。
-4. `.then()` 函数链式调用我们的 promise，当 promise resolve 的时候，里面的代码就会执行。
-5. 最后面我们链了一个 `.catch()` 函数，当 promise rejected 才会执行。
-
-这就注册了一个 service worker，它工作在 worker context，所以没有访问 DOM 的权限。在正常的页面之外运行 service worker 的代码来控制它们的加载。
+1. `if` 代码块进行特性检测测试，以确保在尝试注册 service worker 之前，该特性是被支持的。
+2. 接着，我们使用 [`ServiceWorkerContainer.register()`](/zh-CN/docs/Web/API/ServiceWorkerContainer/register) 函数来注册站点的 service worker。service worker 代码只是一个驻留在我们的 app 内的一个 JavaScript 文件（注意，这个文件的 URL 是相对于源（origin）的，而不是相对于引用它的那个 JS 文件）。
+3. `scope` 参数是可选的，并且可以用来指定你想要 service worker 控制的子作用域。在这个例子中，我们指定了 `'/'`，其表示 app 的源（origin）下的所有内容。如果你留空的话，它的默认值也是这个，但是我们在这里指定它是为了更明确的阐述我们的目的。
 
-单个 service worker 可以控制很多页面。每个你的 scope 里的页面加载完的时候，安装在页面的 service worker 可以控制它。牢记你需要小心 service worker 脚本里的全局变量：每个页面不会有自己独有的 worker。
+这就注册了一个 service worker，它工作在 worker 上下文，所以没有访问 DOM 的权限。
 
-> **备注：** 你的 service worker 函数像一个代理服务器一样，允许你修改请求和响应，用他们的缓存替代它们等等。
+单个 service worker 可以控制很多页面。每个你的作用域（scope）里的页面加载完的时候，安装在页面的 service worker 就可以控制它。牢记你需要小心 service worker 脚本里的全局变量：每个页面不会有自己独有的 worker。
 
-> **备注：** 关于 service workers 一个很棒的事情就是，如果你用像上面一样的浏览器特性检测方式检测发现浏览器并不支持 SW，你还是可以正常地在线使用页面。与此同时，如果你在一个页面上同时使用 AppCache 和 SW , 不支持 SW 但是支持 AppCache 的浏览器，可以使用 AppCache，如果都支持的话，则会采用 SW
+> **备注：** 关于 service worker 一个很棒的事情就是，如果你像我们上面做的那样使用特性检测，发现浏览器并不支持 service worker，但是它还是可以正常地以预期的方式在线使用你的 app。
 
 #### 为什么我的 service worker 注册失败了？
 
 可能是如下的原因：
 
-1. 你没有在 HTTPS 下运行你的程序
-2. service worker 文件的地址没有写对— 需要相对于 origin , 而不是 app 的根目录。在我们的例子例，service worker 是在 `https://mdn.github.io/sw-test/sw.js`，app 的根目录是 `https://mdn.github.io/sw-test/`。应该写成 `/sw-test/sw.js` 而非 `/sw.js`.
-3. service worker 在不同的 origin 而不是你的 app 的，这是不被允许的。
-
-![](important-notes.png)
-
-也请注意：
-
-- service worker 只能抓取在 service worker scope 里从客户端发出的请求。
-- 最大的 scope 是 service worker 所在的地址
-- 如果你的 service worker 被激活在一个有 `Service-Worker-Allowed` header 的客户端，你可以为 service worker 指定一个最大的 scope 的列表。
-- 在 Firefox, Service Worker APIs 在用户在 [private browsing mode](https://support.mozilla.org/en-US/kb/private-browsing-use-firefox-without-history) 下会被隐藏而且无法使用。
+- 你没有在 HTTPS 下运行你的程序。
+- service worker 文件的路径没有写对——需要相对于源（origin），而不是 app 的根目录。在我们的示例中，worker 是在 `https://mdn.github.io/sw-test/sw.js`，app 的根目录是 `https://mdn.github.io/sw-test/`。但是路径需要写成 `/sw.js`。
+- 也不允许你的 app 指向不同源（origin）的 service worker。
+- service worker 只能在 service worker 作用域内捕获客户端发出的请求。
+- service worker 最大的作用域是 worker 所在的位置（换句话说，如果脚本 `sw.js` 位于 `/js/sw.js` 中，默认情况下它只能控制 `/js/` 下的 URL）。可以使用 [`Service-Worker-Allowed`](/zh-CN/docs/Web/HTTP/Header/Service-Worker-Allowed) 标头指定 worker 的最大作用域列表。
+- 在 Firefox 中，Service Worker API 是隐藏的，在用户处于[无痕浏览模式](https://support.mozilla.org/zh-CN/kb/private-browsing-use-firefox-without-history)时、禁用历史记录或者关闭 Firefox 清除 cookie 时，无法使用。
```

Comment:
https://stackoverflow.com/questions/49539306/firefox-service-worker-securityerror-domexception-the-operation-is-insecure

```suggestion
- 在 Firefox 中，若用户处于[无痕浏览模式](https://support.mozilla.org/zh-CN/kb/private-browsing-use-firefox-without-history)、禁用了历史记录或者启用了在 Firefox 关闭时清除历史记录，Service Worker API 将被隐藏而无法使用。
```